src/matrices.rs

```diff
@@ -530,4 +555,14 @@ mod tests {
             r"\begin{bmatrix}2 & 4 & 6\\8 & 10 & 12\end{bmatrix}"
         );
     }
+
+    #[test]
+    fn test_simple_exponentiation() {
+        let m = im![1, 1; 1, 0];
+        assert_eq!(m.checked_pow(0).unwrap(), im![1, 0; 0, 1]);
+        assert_eq!(m.checked_pow(1).unwrap(), im![1, 1; 1, 0]);
+        assert_eq!(m.checked_pow(2).unwrap(), im![2, 1; 1, 1]);
+        assert_eq!(m.checked_pow(9).unwrap(), im![55, 34; 34, 21]);
+        assert_eq!(m.checked_pow(10).unwrap(), im![89, 55; 55, 34]);
+    }
```

Comment:
I would add a test to check `Rational64` as well

Reply:
sufficient?

Reply:
Yeah!